pipeline/pipeline_runner.go

```diff
@@ -158,7 +158,7 @@ type PipelinePack struct {
 // Returns a new PipelinePack pointer that will recycle itself onto the
 // provided channel when a message has completed processing.
 func NewPipelinePack(recycleChan chan *PipelinePack) (pack *PipelinePack) {
-	msgBytes := make([]byte, message.MAX_MESSAGE_SIZE)
+	msgBytes := make([]byte, 0, message.MAX_MESSAGE_SIZE)
```

Comment:
This worries me a bit. `pack.Zero()` sets MsgBytes length to the full capacity, and our code generally copies into MsgBytes and then resizes down, rather than resizing up and then copying. I'm okay w/ changing this, but I'd rather do it in a separate PR, since it requires digging into all of the places where we set pack.MsgBytes to make sure we don't silently truncate b/c the length is too short.